lib/network.go

```diff
@@ -2194,7 +2194,7 @@ func (msg *MsgDeSoHeader) EncodeHeaderVersion2(preSignature bool) ([]byte, error
 
 	// TstampNanosSecs: this field can be encoded to take up to the full 64 bits now
 	// that MsgDeSoHeader version 2 does not need to be backwards compatible.
-	retBytes = append(retBytes, UintToBuf(msg.TstampNanoSecs)...)
+	retBytes = append(retBytes, IntToBuf(msg.TstampNanoSecs)...)
```

Comment:
I don't think it's safe to just change this, but I'm not 100% sure. Basically I'd opt to continue encoding it with UintToBuf after casting to uint64. We should code defensively to make sure the value is never negative. I'm worried that this will mess with syncing since headers won't match.